torch/_refs/__init__.py

```diff
@@ -2846,6 +2848,55 @@ def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLi
     return where(mask, prims.to_dtype(value, a.dtype), a)
 
 
+def allclose(
+    a: TensorLikeType,
+    b: TensorLikeType,
+    rtol: Optional[Union[bool, int, float]] = None,
+    atol: Optional[Union[bool, int, float]] = None,
+    equal_nan: Optional[bool] = None,
+) -> bool:
+    """
+    Reference implementation of torch.allclose
+    """
+    if rtol is None:
+        rtol = 1e-05
+
+    if atol is None:
+        atol = 1e-08
+
+    if equal_nan is None:
+        equal_nan = False
+
+    # Checks similar to equal, but slightly different to match the op behavior
+    utils.check_same_device(a, b, allow_cpu_scalar_tensors=True)
+    utils.check_same_dtype(a, b)
+
+    # Shape check
+    if a.ndim != b.ndim:
+        return False
```

Comment:
Following on what @Lezcano said -- why not let `isclose` handle this behavior?

I would expect an implementation for `allclose` to follow the existing implementation since `allclose` is a composite operation:

https://github.com/pytorch/pytorch/blob/1d90d6ee60f2e1850556fd26d7ffde950ab992a1/aten/src/ATen/native/TensorCompare.cpp#L223